elytron/ELY-2320-integrity-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,149 @@
+
+To store the KeyPair, a key store will be created and the FileSystem realm will utilize the key store to access the PublicKey and PrivateKey to verify filesystem integrity.
+
+For the default out of box configuration for WildFly, the goal is for the FileSystemSecurityRealm to have integrity enabled.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/ELY-2320[ELY-2320]
+* https://issues.redhat.com/browse/WFCORE-5859[WFCORE-5859]
+* https://issues.redhat.com/browse/WFLY-16179[WFLY-16179]
+
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1892[EAP7-1892]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+- Widlfly, Wildfly Elytron, Wildfly Core
+
+=== Other Interested Projects
+N/A
+
+== Requirements
+=== Integrity of Filesystem Realm
+
+Filesystem Integrity is a planned upcoming change, but will not be added until after WildFly 26.1 is released
+
+The two approaches being considered are between signing each identity file, and just calculating the checksum of each identity file.
+
+The advantage of using just a straightforward checksum using a hash algorithm is that it will not be resource heavy, as an algorithm such as SHA256 or MD5 is very efficient, although it comes with a caveat that it is possible to produce collisions (false positivies) due to the lack of complexity of the algorithm. The disadvantage of using a simple checksum is that if an authorized user were to gain access to the master list of checksums that is used to verify each identity when accessed, they could change the checksum rendering that identity to be useless, until that checksum is amended. Is it also possible for the unauthorized user to modify the identity file, calculate the new checksum, and update the checksum list with an updated one. This would be very detrimental as we have no way of knowing that the file has been modified and is no longer valid.
+
+If the approach of signing a file was chosen, it would require an asymmetric key pair, a public key and private key. This would involve a key store and a method to ensure only the correct party has access to the private key.
+
+It could be made that the public key is available for everyone to use. This means anyone can decrypt the signature to get the hash of the identity, but only people with access to the key store for the realm would have access to the private key that encrypts the hash of each identity. This way an unauthorized user could view the hash via the public key, but cannot change the hash/signature of the identity.
+
+The approach that is taken is to use the key pair file signature approach. To do so, ever time data is to be written or fetched by an identity, first the filesystem realm will verify the signature with identity file. If this signature ever does not match the identity, then the filesystem realm integrity has been compromised.
```

Comment:
s/by an identity/for an identity

Reply:
s/If this signature ever does not match the identity/If signature verification fails